monix-execution/shared/src/main/scala/monix/execution/CancelableFuture.scala

```diff
@@ -31,7 +32,7 @@ import scala.util.control.NonFatal
 /** Represents an asynchronous computation that can be canceled
   * as long as it isn't complete.
   */
-sealed abstract class CancelableFuture[+A] extends Future[A] with Cancelable { self =>
+sealed abstract class CancelableFuture[+A](isolatedCtx: Local.Context) extends Future[A] with Cancelable { self =>
```

Comment:
Temporary solution to see if it's worth pursuing - if it works, I'll hide everything